test/Table_Tests/src/Common_Table_Operations/Join/Join_Spec.enso

```diff
@@ -479,9 +479,9 @@ add_join_specs suite_builder setup =
             Problems.assume_no_problems r2
 
             if setup.supports_custom_objects then
-                t1 = table_builder [["X", [My_Type.Value 1 2, 2.0, 2]], ["Y", [10, 20, 30]]]
-                t2 = table_builder [["Z", [2.0, 1.5, 2.0]], ["W", [1, 2, 3]]]
-                r3 = t1.join t2 join_kind=Join_Kind.Inner on=(Join_Condition.Equals "X" "Z") on_problems=..Report_Warning
+                t8 = table_builder [["X", [My_Type.Value 1 2, 2.0, 2]], ["Y", [10, 20, 30]]]
```

Comment:
I don't understand the requirement that you cannot override a binding from parent scope in the branches. The true branch and else branch is nothing more than a block of expressions, right? The following code:
```
foo ~action =
    action

main =
    x = 42
    foo <|
        x = 23
        x
```
prints 23, so it let's me override the `x` from the parent scope as well.

Is this the desired state or is this still WIP?